hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmCriteriaNodeBuilder.java

```diff
@@ -923,12 +937,123 @@ public <N extends Number> SqmExpression<N> abs(Expression<N> x) {
 		);
 	}
 
+	@Override
+	public JpaExpression<Duration> duration(long magnitude, TemporalUnit unit) {
+		return new SqmToDuration<>(
+				literal( magnitude ),
+				new SqmDurationUnit<>( unit, getLongType(), this ),
+				getTypeConfiguration().standardBasicTypeForJavaType( Duration.class ),
+				this
+		);
+	}
+
+	@Override
+	public JpaExpression<Long> magnitude(TemporalUnit unit, Expression<Duration> duration) {
+		return new SqmByUnit(
+				new SqmDurationUnit<>( unit, getLongType(), this ),
+				(SqmExpression<Duration>) duration,
+				getLongType(),
+				this
+		);
+	}
+
+	@Override
+	public JpaExpression<Duration> add(Expression<Duration> x, Expression<Duration> y) {
```

Comment:
I don't have a good name suggestion, but it feels "wrong" to me to reserve/use the names `add`/`subtract` (considering JVM erasure) for the addition/subtraction of two durations. For numerics, JPA claimed `sum`/`diff`. 

Adding a duration/interval to a temporal needs dedicated methods as well due to erasure and the lack of a common type between `Temporal` and `Duration`.

For the lack of better ideas, how about `durationAdd(Expression<Duration> x, Expression<Duration> y)` and `temporalAdd(Expression<? extends Temporal> x, Expression<Duration> y)`?